nntrainer/models/neuralnet.cpp

```diff
@@ -257,7 +257,6 @@ int NeuralNetwork::initialize() {
       l.setInputBuffers(in_out);
     }
   }
-  setBatchSize(batch_size);
```

Comment:
as far as I remember, this was added 
from https://github.com/nnstreamer/nntrainer/pull/649#discussion_r508173637

If a user is prohibited to call `forward` or `backward`.
I think this changes is fine.

If she is allowed to, either she has to explicitly call `setBatchSize` or `setBatchSize` should be done at `initailize` phase.
I think some of our application doesn't follow this and might break.

Reply:
`setBatchSize` currently creates an overhead of memory being allocated with pooling layer, with not the final batch size.
I will resolve this with an additional interface in the tensor manager for allocating extra memory for a layer.

Reply:
Cool